src/Stratis.Features.FederatedPeg/Wallet/MultiSigTransactions.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    public class MultiSigTransactions : ICollection<TransactionData>, ITransactionDataObserver
```

Comment:
By keeping this as ICollection<> and looping through it in many places in the FWM, we lose a lot of the benefit of this class. eg. `TransactionData spentTransaction = this.Wallet.MultiSigAddress.Transactions.SingleOrDefault(t => (t.Id == spendingTransactionId) && (t.Index == spendingTransactionIndex));`
and 
`foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions.ToList())`

Were you planning to get to these?

Reply:
I've addressed the first now. The second is only called once at startup.